service/src/main/java/com/derpgroup/quip/manager/QuipManager.java

```diff
@@ -101,47 +119,56 @@ protected void doGoodbyeRequest(VoiceInput voiceInput,
   protected void doConversationRequest(VoiceInput voiceInput,
       SsmlDocumentBuilder builder) {
 
+    Map<String,String> messageMap = voiceInput.getMessageAsMap();
     QuipMetadata metadata = (QuipMetadata) voiceInput.getMetadata();
-
     String messageSubject = voiceInput.getMessageSubject();
 
+    switchOnSubject(messageSubject, messageMap, builder, metadata);
+  }
+
+  public void switchOnSubject(String messageSubject, Map<String,String> messageMap, SsmlDocumentBuilder builder, QuipMetadata metadata) {
     switch (messageSubject) {
     case "COMPLIMENT":
-      doComplimentRequest(voiceInput, builder, metadata);
+      doComplimentRequest(messageMap, builder, metadata);
       break;
     case "INSULT":
-      doInsultRequest(voiceInput, builder, metadata);
+      doInsultRequest(messageMap, builder, metadata);
       break;
     case "BACKHANDED_COMPLIMENT":
-      doBackhandedComplimentRequest(voiceInput, builder, metadata);
+      doBackhandedComplimentRequest(messageMap, builder, metadata);
       break;
     case "WINSULT":
-      doWinsultRequest(voiceInput, builder, metadata);
-      break;
-    case "HELP":
-      doHelpRequest(voiceInput, builder);
+      doWinsultRequest(messageMap, builder, metadata);
       break;
     case "WHO_BUILT_YOU":
-      doWhoBuiltYouRequest(voiceInput, builder);
+      doWhoBuiltYouRequest(messageMap, builder, metadata);
       break;
     case "WHAT_DO_YOU_DO":
-      doWhatDoYouDoRequest(voiceInput, builder);
+      doWhatDoYouDoRequest(messageMap, builder, metadata);
       break;
     case "FRIENDS":
-      doFriendsRequest(voiceInput, builder);
+      doFriendsRequest(messageMap, builder, metadata);
       break;
     case "WHO_IS":
-      doWhoIsRequest(voiceInput, builder);
+      doWhoIsRequest(messageMap, builder, metadata);
+      break;
+    case "ANOTHER":
+      doAnotherRequest(messageSubject, messageMap, builder, metadata);
       break;
     default:
       builder.text("Unknown request type '" + messageSubject + "'.");
     }
   }
 
-  private void doWhoIsRequest(VoiceInput voiceInput, SsmlDocumentBuilder builder) {
-    QuipMetadata metadata = (QuipMetadata) voiceInput.getMetadata();
+  private void doAnotherRequest(String messageSubject, Map<String, String> messageMap, SsmlDocumentBuilder builder, QuipMetadata metadata) {
+    //this has its own method in case we want to do things like logging
+    ConversationHistoryEntry entry = ConversationHistoryUtils.getLastNonMetaRequestBySubject(metadata.getConversationHistory(), new HashSet<String>(Arrays.asList(metaRequestSubjects)));
+    switchOnSubject(entry.getMessageSubject(), messageMap, builder, metadata);
```

Comment:
We need a null check here. If you ask for another without having a prior, it blows up.
"Alexa, ask Insultibot for another"